rbc/external.py

```diff
@@ -1,38 +1,65 @@
 # This code has heavily inspired in the numba.extending.intrisic code
-
+from typing import Dict, List
+from collections import defaultdict
 from numba.core import extending, funcdesc, types, typing
 from rbc.typesystem import Type
 from rbc.targetinfo import TargetInfo
 
 
 class External:
     @classmethod
-    def fromobject(cls, signature, name: str = None):
+    def fromobject(cls, *args, typing=True, lowering=True):
         """
         Parameters
         ----------
         signature : object (str, ctypes function, python callable, numba function)
             Any object convertible to a Numba function via Type.fromobject(...).tonumba()
         name : str
             The name of the external function
+        typing : bool
+            Indicates if External should do typing or not. Default is True
+        lowering: bool
+            Indicates if External should do lowering or not. Default is True. Except for
+            very specific cases, typing and lowering should be True
         """
         # Make inner function for the actual work
         target_info = TargetInfo.dummy()
+        ts = defaultdict(list)
```

Comment:
Notice that all signatures will be with respect to dummy target info. This is problematic because there may exist signatures that contain types that actually depend on the device or on the actual target.

For example, we have `sizeof(long) == 32` on Windows. Eventhough, Type.fromobject does not evaluate `long` when using dummy target, `Type` does not reevaluate the signature when under specific target info (maybe it should but not here).

To avoid any issues that the above might trigger, I suggest storing the original `signature` objects in `ts` rather than the `Type` instances. This will duplicate `Type.fromobject` calls (until we'll support `Type.reevaluate`) but at least the `signature` objects will be transformed to `Type` instances within the correct target info.

Reply:
Done!